cordova-lib/src/cordova/plugin.js

```diff
@@ -305,6 +289,49 @@ module.exports = function plugin(command, targets, opts) {
     });
 };
 
+function determinePluginTarget(projectRoot, cfg, target, fetchOptions) {
+    var parts = target.split('@');
+    var id = parts[0];
+    var version = parts[1];
+
+    // If no version is specified, retrieve the version (or source) from config.xml
+    if (version || cordova_util.isUrl(id) || cordova_util.isDirectory(id)) {
+        return Q(target);
+    } else {
+        events.emit('verbose', 'No version specified, retrieving version from config.xml');
+        var ver = getVersionFromConfigFile(id, cfg);
+
+        if (cordova_util.isUrl(ver) || cordova_util.isDirectory(ver)) {
+            return Q(ver);
+        } else if (ver) {
+            // If version exists in config.xml, use that
+            return Q(id + '@' + ver);
+        } else {
+            // If no version is given at all and we are fetching from npm, we
+            // can attempt to use the Cordova dependencies the plugin lists in
+            // their package.json
+            var shouldUseNpmInfo = !fetchOptions.searchpath && !fetchOptions.noregistry;
+
+            if(shouldUseNpmInfo) {
+                events.emit('verbose', 'No version given in config.xml, attempting to use plugin engine info');
+            }
+
+            return (shouldUseNpmInfo ? registry.info([id]) : Q({}))
```

Comment:
Right! I wrote this before I was aware that jasmine could mock function calls. As you said, I want to get this in today so I'm going to leave it out for now since that would be a big change to the tests..